### PR TITLE
Added Firefox OS device API lookup

### PR DESCRIPTION
**Pull request by tagawa**
**What does your Instant Answer do?**
Shows information for individual Firefox OS device APIs

**What problem does your Instant Answer solve (Why is it better than organic links)?**
Goes straight to information for a particular API rather than to the table of all APIs.

**What is the data source for your Instant Answer? (Provide a link if possible)**
Mozilla Developer Network:
https://developer.mozilla.org/en-US/Apps/Reference/Firefox_OS_device_APIs

**Why did you choose this data source?**
They created the APIs

**Are there any other alternative (better) data sources?**
No.

**What are some example queries that trigger this Instant Answer?**
fxos alarm api
firefoxos

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Firefox OS app developers

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
https://duck.co/ideas/idea/6284/firefox-os-api-lookup

**Which existing Instant Answers will this one supercede/overlap with?**
None

**Are you having any problems? Do you need our help with anything?**
Yes. How do I get the title or subtitle to use the URL in the structured answer?

**Where did you hear about DuckDuckHack? (For first time contributors)**

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
fxos-apis

## Checklist
Please place an 'X' where appropriate.

```
- [x] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- [x] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
- [x] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- [x] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [x] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Ubuntu 14.04
        - [x] Chromium
        - [x] Firefox
        - [x] Opera
        - [x] Vivaldi

    - Android
        - [x] Firefox
        - [x] Native Browser - doesn't seem to display for some reason (Android 4.4.4)
        - [] Google Chrome
        - [x] Opera

```

IA Page: https://duck.co/ia/view/firefox_os

### CODE REVIEW

**Comment by tagawa** [comment]
Note that there is data in the ```apis.yml``` that is not used. I'd like to improve this to include that data later on.

**Comment by tagawa** [comment]
Ignore the last comment. I changed the template group to ```info``` and that a) fixed the URL problem, and b) enabled me to put the unused data in an infobox. Double win!

fxos-apis

**Comment by tagawa** [comment]
By the way, if you know of any easy way to stop the permission name being capitalised (i.e. so it says ```Manifest Permission: alarm```) please let me know.

**Comment by moollaza** [comment]
@tagawa this is pretty awesome! -- Is this data likely to change? It seems like a Fathead might be ideal for this kind of info? Seeing as you've used the Info template (which is essentially what Fatheads use) you should be able to recreate the design.

Also, did you use a script to produce the YAML or was it done by hand?

**Comment by tagawa** [comment]
Thanks @moollaza. I lovingly crafted the YAML by hand adding a couple of things such as the aliases.

The source data is just a single table in [the API page](https://developer.mozilla.org/en-US/Apps/Reference/Firefox_OS_device_APIs). I considered a Fathead but I checked with open@ddg and we thought a Goodie would be OK because the data doesn't seem to be changing much, if at all.

File: lib/DDG/Goodie/FirefoxOS.pm [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    $key = trim $a . ' ' . trim $b;
+    
+    my $api = $apis->{$key};
+
+    return unless ($api);
+    
+    return $api->{permission},
+      structured_answer => {
+        id => "firefox_os",
+        name => "Firefox OS",
```
Comment (moollaza) [inline lib/DDG/Goodie/FirefoxOS.pm]: Let's change this to `About` so it acts like our Fatheads

**Comment by moollaza** [comment]
@tagawa excellent work! Let's change the `name` and then this should be good for merge :+1: 

Ultimately, I think it would be best if this was converted to a Fathead given it's nature and ideally we could use a script to generate the YAML if possible so it's simply to maintain.

**Comment by tagawa** [comment]
Thanks @moollaza. I've been thinking about trying my hand at a Fathead so converting this would be a good first step, and better for long-term maintainability as you say.

**Comment by moollaza** [comment]
LGTM :+1:

File: t/FirefoxOS.t [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+use DDG::Test::Goodie;
+
+zci answer_type => "firefoxos";
+zci is_cached   => 1;
+
+#Structured answer template data
+my $templateData = {
+            data => '-ANY-',
+            id => "firefox_os",
+            name => "Firefox OS",
```
Comment (moollaza) [inline t/FirefoxOS.t]: Looks like this needs to be updated as well. I'll merge and fix :+1:

Comment (tagawa) [inline t/FirefoxOS.t]: Oops, forgot about that. Thanks.

**Comment by moollaza** [comment]
@tagawa congrats! It's live: https://duckduckgo.com/?q=firefox+os+alarm+api&ia=about

**Comment by tagawa** [comment]
Cool! Thanks for letting me know :tada:

**Pull request merged**